controllers/category.controller.js

```diff
@@ -0,0 +1,96 @@
+const { extend } = require("lodash");
+const mongoose = require("mongoose");
+const { Category } = require("../models/Category.model");
+const { Video } = require("../models/Video.model");
+const getAllCategories = async (req, res) => {
+  try {
+    const allCategories = await Category.find({});
+    res.status(200).json({
+      success: true,
+      message: "All Categories",
+      categories: allCategories,
+    });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      message: "Error in getting categories",
+      errMessage: err.errMessage,
+    });
+  }
+};
+const addACategory = async (req, res) => {
+  const categoryDetails = req.body;
+  const newCategory = new Category(categoryDetails);
+  let savedCategory;
+  try {
+    savedCategory = await newCategory.save();
+    res.status(201).json({
+      success: true,
+      message: "Added new category",
+      category: savedCategory,
+    });
+  } catch (err) {
+    res.status(500).json({
+      success: false,
+      message: "Error in adding category",
+      errMessage: err,
+    });
+  }
+};
+const updateCategory = async (req, res) => {
+  let category = req.category;
+  const updateCategory = req.body;
+  category = extend(category, updateCategory);
+  try {
+    const updatedCategory = await category.save();
+    res.status(204).json({
```

Comment:
Here the status should be 201 for `post` update operation
```suggestion
    res.status(201).json({
```